### PR TITLE
new(DateTime, DateTimeRange, DateTimeSelect, Price, PriceComparison, PriceGroup): Add empty fallback if an invalid value is provided.

### PR DESCRIPTION
**Pull request by stefhatcher**
to: @milesj @stefhatcher @hayes @alecklandgraf

## Description

handle bad values like `"[Hidden]"` being added to things expected timestamps or numbers representing an amount.

## Motivation and Context

caused components to crash

## Testing

- storybook
- new tests
- old tests passing

## Screenshots

Core : DateTime - Fallback when an invalid date value is provided   ⋅ Storybook 2020-04-30 12-39-34
Core : DateTimeRange - Fallback when an invalid date values are provided   ⋅ Storybook 2020-04-30 12-40-05
Core : DateTimeSelect - Fallback to today ' s date if value is invalid ⋅ Storybook 2020-04-30 12-40-29
Core : Price - Fallback when an invalid amount is provided   ⋅ Storybook 2020-04-30 12-41-09
Core : PriceComparison - Fallback when invalid amounts are provided   ⋅ Storybook 2020-04-30 12-41-27
Core : PriceGroup - Fallback when invalid amounts are provided   ⋅ Storybook 2020-04-30 12-42-00

## Checklist

- My code follows the style guide of this project.
- I have updated or added documentation accordingly.
- I have read the CONTRIBUTING document.

### CODE REVIEW

File: packages/core/src/components/DateTimeSelect/index.tsx [hunk 1]
```diff
@@ -83,18 +88,30 @@ export class DateTimeSelect extends React.Component<
...
+      const date =
+        createDateTime(value, {
+          locale,
+          timezone,
+        }) ?? this.getNowDate();
 
       this.setState({
         date,
-        meridiem: date.get('hour') <= 11 ? 'am' : 'pm',
+        meridiem: date && date.get('hour') <= 11 ? 'am' : 'pm',
```
Comment (milesj) [inline packages/core/src/components/DateTimeSelect/index.tsx:99]: `date?.get`

File: packages/core/src/components/Price/index.tsx [hunk 2]
```diff
@@ -43,14 +43,20 @@ function Price({
   let currency = baseCurrency;
   let micros = baseMicros;
 
-  if (baseAmount === undefined || baseAmount === null) {
+  if (
+    baseAmount === undefined ||
+    baseAmount === null ||
+    (typeof baseAmount === 'string' && isNaN(Number(baseAmount)))
```
Comment (milesj) [inline packages/core/src/components/Price/index.tsx:49]: `Number.isNaN`

File: packages/core/src/components/PriceComparison/index.tsx [hunk 3]
```diff
@@ -26,22 +27,29 @@ function PriceComparison({
   const hasPriceUSD = amountUSD !== null && amountUSD !== undefined;
   const targetCurrency = amount && typeof amount === 'object' ? amount.currency : currency!;
 
+  if (
+    (typeof amount === 'string' && isNaN(Number(amount))) ||
```
Comment (milesj) [inline packages/core/src/components/PriceComparison/index.tsx:31]: `Number.isNan`

File: packages/core/src/components/PriceGroup/index.tsx [hunk 4]
```diff
@@ -30,6 +31,19 @@ export default function PriceGroup({ amounts, divider = ', ', ...restProps }: Pr
   // Loop through and generate the prices
   const output: JSX.Element[] = [];
 
+  let invalidAmounts = 0;
+  currencies.forEach(currency => {
+    const amount = amounts[currency];
+
+    if (typeof amount === 'string' && isNaN(Number(amount))) {
```
Comment (milesj) [inline packages/core/src/components/PriceGroup/index.tsx:38]: Same here.

File: packages/core/src/components/DatePickerInput/index.tsx [hunk 5]
```diff
@@ -71,6 +71,7 @@ export default class DatePickerInput extends React.Component<
     // Update the parent form with the selected value.
     // We also don't have a real event object, so fake it.
     this.props.onChange(
+      // @ts-ignore null check happened above
```
Comment (milesj) [inline packages/core/src/components/DatePickerInput/index.tsx:74]: Use `!` instead?

File: packages/core/src/components/DatePickerInput/index.tsx [hunk 6]
```diff
@@ -103,7 +105,7 @@ export default class DatePickerInput extends React.Component<
       locale: locale ?? this.props.locale,
       noTime: true,
       noTimezone: true,
-    });
```
Comment (milesj) [inline packages/core/src/components/DatePickerInput/index.tsx:106]: This seems like a trap.

Comment (stefhatcher) [reply packages/core/src/components/DatePickerInput/index.tsx:106]: yeah it made ts happy 😬

Comment (milesj) [reply packages/core/src/components/DatePickerInput/index.tsx:106]: Is the original type nullable?

Comment (stefhatcher) [reply packages/core/src/components/DatePickerInput/index.tsx:106]: ah, i can remove this! an artifact of a different approach

File: packages/core/src/components/Price/index.tsx (hunk 2 above)
Comment (stefhatcher) [reply packages/core/src/components/Price/index.tsx:49]: @milesj going to leave these as is, it's a little cleaner:

`Number.isNaN('[Hidden]')` -> `false`
`Number.isNaN(Number('[hidden]'))` -> `true`
`isNaN(Number('[Hidden]'))` -> `true`

**Pull request merged**